File: src/api/domain/category_suggestion.go
package domain

import "go.mongodb.org/mongo-driver/bson/primitive"

type CategorySuggestion struct {
	Id             primitive.ObjectID `json:"id" bson:"_id" required:"false"`
	CategoryName   string             `json:"name" bson:"name" required:"false"`
	HiddenCategory bool               `json:"hidden" bson:"hidden" required:"false"`
	CategoryIcon   string             `json:"iconClass" bson:"icon" required:"false"`
	Status         Status             `json:"status" bson:"status" required:"false"`
}

type Status string

const (
	Pending  Status = "waiting_for_approval"
	Approved Status = "approved"
	Rejected Status = "rejected"
)

File: src/ui/src/app/domain/category-suggestion.ts
import {Category} from "./category";

export class CategorySuggestion {
  id: string;
  category: Category;
  status: SuggestionStatus;


  constructor(category: Category, status: SuggestionStatus) {
    this.category = category;
    this.status = status;
  }
}

export enum SuggestionStatus {
  APPROVED,
  WAITING_FOR_APPROVAL,
  REJECTED
}
